test_util/aws.py

```diff
@@ -87,7 +88,7 @@ def create(cls, stack_name, template_url, public_agents, private_agents,
             'PublicSlaveInstanceCount': str(public_agents),
             'SlaveInstanceCount': str(private_agents)}
         stack = boto_wrapper.create_stack(stack_name, template_url, parameters)
-        return cls(stack.stack.stack_id, boto_wrapper)
+        return cls(stack.stack.stack_name, boto_wrapper)
```

Comment:
This is from Boto being stupid: when you name a stack from stack_name, you can then call stack.stack_name and you get the StackName or you can call stack.stack_id and get the physical resource ID. If you create it from the physical resource ID, then stack.stack_name returns the ID and not the name. So, when I want to find all the stacks the most sensible filter is: 
```
Filter=[{'Name': 'tag:aws:cloudformation:stack-name', 'Values': [self.stack.stack_name]}]
```
which breaks when stack.stack_name returns an ID.

Suggestion? This was the most sensible approach I could find other than setting the stack_name by describing the stack_id and getting the tag (which seemed excessive)